tap_facebook/__init__.py

```diff
@@ -62,6 +64,37 @@ def transform_datetime_string(dts):
         parsed_dt = parsed_dt.astimezone(timezone.utc)
     return singer.strftime(parsed_dt)
 
+class InsightsJobTimeout(Exception):
+    pass
+
+def retry_pattern(backoff_type, exception, **wait_gen_kwargs):
+    def log_retry_attempt(details):
+        _, exception, _ = sys.exc_info()
+        LOGGER.info(exception)
+        LOGGER.info('Caught retryable error after {tries} tries. Waiting {wait:.0f} more seconds then retrying...'.format(**details))
```

Comment:
A couple things here:

1. I don't think it's good to have one function take a map and then pass it in as kwargs to another function. Here we're depending on the caller, which is another library, to pass a map containing "wait" and "tries" in to this function. If it ever didn't do that for some reason the error would surface in a weird way.
2. I would recommend uinsg `%s` and letting the logger format the string rather than using `{}` and calling format explicitly. That's what PyPi recommends doing. It's not important in this case, but I think it's a good habit to get into. It makes it so if the arguments to the format string are complex they're only actually stringified if the logging is turned on for that level.